source/iterator/snapshot_iterator.go

```diff
@@ -0,0 +1,386 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	marketoclient "github.com/rustiever/conduit-connector-marketo/marketo-client"
+	"github.com/rustiever/conduit-connector-marketo/source/position"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// is the maximum number of days between two snapshots. If the gap between two snapshots is greater than this number,
+	// API will return an error. This is limitation of the API.
+	MaximumDaysGap = 744 // 31 days in Hours
+)
+
+var (
+	InitialDate time.Time // holds the initial date of the snapshot
```

Comment:
Why not make it part of SnapshotIterator?

Reply:
this variable was initially with SnapshotIterator, later moved as Global var for testing. if it is in SnapshotIterator then tests taking too much time. so for testing `InitialDate` will be current datetime

Reply:
If the variable is part of SnapshotIterator, then you can use a different value for it in testing, no? Introducing a global variable only to make tests work feels like something which can be improved.

Reply:
will be handled in next PR @hariso